src/plugins/dashboard/server/content_management/dashboard_storage.ts

```diff
@@ -32,34 +49,315 @@ const searchArgsToSOFindOptions = (
+    const { value: dataToLatest, error: dataError } = transforms.create.in.data.up<
+      DashboardAttributes,
+      DashboardAttributes
+    >(data);
+    if (dataError) {
+      throw Boom.badRequest(`Invalid data. ${dataError.message}`);
+    }
+
+    const { value: optionsToLatest, error: optionsError } = transforms.create.in.options.up<
+      DashboardCreateOptions,
+      DashboardCreateOptions
+    >(options);
+    if (optionsError) {
+      throw Boom.badRequest(`Invalid options. ${optionsError.message}`);
+    }
+
+    const soAttributes = itemAttrsToSavedObjectAttrs(dataToLatest);
+
+    // Save data in DB
+    const savedObject = await soClient.create<DashboardSavedObjectAttributes>(
+      DASHBOARD_SAVED_OBJECT_ID,
+      soAttributes,
+      optionsToLatest
+    );
+
+    const result = {
+      item: savedObjectToItem(savedObject, false),
+    };
+
+    const validationError = transforms.create.out.result.validate(result);
+    if (validationError) {
+      if (this.throwOnResultValidationError) {
+        throw Boom.badRequest(`Invalid response. ${validationError.message}`);
+      } else {
+        this.logger.warn(`Invalid response. ${validationError.message}`);
+      }
+    }
+
+    // Validate DB response and DOWN transform to the request version
+    const { value, error: resultError } = transforms.create.out.result.down<
+      CreateResult<DashboardItem>
+    >(
+      result,
+      undefined, // do not override version
+      { validate: false } // validation is done above
+    );
+
+    if (resultError) {
+      throw Boom.badRequest(`Invalid response. ${resultError.message}`);
+    }
+
+    return value;
+  }
+
+  async update(
+    ctx: StorageContext,
+    id: string,
+    data: DashboardAttributes,
+    options: DashboardUpdateOptions
+  ): Promise<DashboardUpdateOut> {
+    const transforms = ctx.utils.getTransforms(cmServicesDefinition);
+    const soClient = await savedObjectClientFromRequest(ctx);
+
+    // Validate input (data & options) & UP transform them to the latest version
+    const { value: dataToLatest, error: dataError } = transforms.update.in.data.up<
+      DashboardAttributes,
+      DashboardAttributes
+    >(data);
+    if (dataError) {
+      throw Boom.badRequest(`Invalid data. ${dataError.message}`);
+    }
+
+    const { value: optionsToLatest, error: optionsError } = transforms.update.in.options.up<
+      DashboardUpdateOptions,
+      DashboardUpdateOptions
+    >(options);
+    if (optionsError) {
+      throw Boom.badRequest(`Invalid options. ${optionsError.message}`);
+    }
+
+    const soAttributes = itemAttrsToSavedObjectAttrs(dataToLatest);
+
+    // Save data in DB
+    const partialSavedObject = await soClient.update<DashboardSavedObjectAttributes>(
+      DASHBOARD_SAVED_OBJECT_ID,
+      id,
+      soAttributes,
+      optionsToLatest
+    );
+
+    const result = {
+      item: savedObjectToItem(partialSavedObject, true),
```

Comment:
You will need a try/catch inside `savedObjectToItem` to catch any errors (for example, JSON.parse can throw). Then `savedObjectToItem` should return a result where `transforms.update.out.result.validate` will return a validationError

Reply:
Addressed in [c505529](https://github.com/elastic/kibana/pull/193067/commits/c50552938e90788b302c6718963a7e9ae2ba9441).